File: P1/E1/Carrera.java
package E1;
import java.util.*;

public abstract class Carrera implements Runnable{

    protected ArrayList<Bicicleta> bicicletas;
    public String nombre;

    int instanteRetirada = 4;

    Carrera(){
        bicicletas = new ArrayList<>();
    }

    public void SetNombre(String nombre){
        this.nombre = nombre;
    }

    public abstract void aniadeBicicleta(Bicicleta bici);

    public abstract void retirada();
}
    

File: P1/E2/Carrera.py
from abc import ABC, abstractmethod #PARA DEFINIR CLASES ABSTRACTAS EN PYTHON.
from typing import List
from Bicicleta import Bicicleta

class Carrera(ABC):
    def __init__(self):
        self.bicicletas = []
        self.nombre: str = ""
        self.instanteRetirada: int = 4

    def setNombre(self, nombre: str):
        self.nombre = nombre

    @abstractmethod
    def run(self):
        pass

    @abstractmethod
    def aniadeBicicleta(self, bici: Bicicleta):
        pass

    @abstractmethod
    def retirada(self):
        pass